src/MasterSite.php

```diff
@@ -275,6 +275,40 @@ public function set_featured_image( $post_id, $post, $update ) {
 		}
 	}
 
+	/**
+	 * Generates hreflang metadata from countries.json template.
+	 */
+	public static function generate_hreflang_meta() {
+
+		$countries = wp_cache_get( 'countries' );
+
+		if ( false === $countries ) {
+			$body      = file_get_contents( get_template_directory() . '/templates/countries.json' );
+			$countries = json_decode( $body, true );
+			wp_cache_set( 'countries', $countries );
```

Comment:
To be safesafesafe and not cache a bad value I'd add
```php
$body      = file_get_contents( get_template_directory() . '/templates/countries.json' );
$countries = json_decode( $body, true );
if ( empty( $countries ) ) {
    return null;
}
wp_cache_set( 'countries', $countries );
```

and change method signature to `public static function generate_hreflang_meta(): ?array {`

Reply:
Thanks! commit amended